File: framework/src/play-java/src/test/scala/play/data/Task.scala
/*
 * Copyright (C) 2009-2016 Lightbend Inc. <https://www.lightbend.com>
 */
package play.data.models

import java.util.Date
import scala.beans.BeanProperty
import scala.annotation.meta.field

class Task {

  type Min = play.data.validation.Constraints.Min @field
  type Required = play.data.validation.Constraints.Required @field
  type I18NConstraint = play.data.validation.TestConstraints.I18Constraint @field
  type DateTime = play.data.format.Formats.DateTime @field

  @Min(10)
  @BeanProperty
  var id: Long = _

  @Required
  @BeanProperty
  var name: String = _

  @BeanProperty
  var done: Boolean = true

  @BeanProperty
  @Required
  @DateTime(pattern = "dd/MM/yyyy")
  var dueDate: Date = _

  @BeanProperty
  var endDate: Date = _

  @BeanProperty
  @I18NConstraint(value = "patterns.zip")
  var zip: String = _

}

File: framework/src/play-java-forms/src/test/java/play/data/Task.java
/*
 * Copyright (C) 2009-2018 Lightbend Inc. <https://www.lightbend.com>
 */

package play.data;

import java.util.Date;

import play.data.format.Formats.DateTime;

import play.data.validation.Constraints;
import play.data.validation.TestConstraints.I18Constraint;
import play.data.validation.TestConstraints.AnotherI18NConstraint;

public class Task {

    @Constraints.Min(10)
    private Long id;

    @Constraints.Required
    private String name;

    private Boolean done = true;

    @Constraints.Required
    @DateTime(pattern = "dd/MM/yyyy")
    private Date dueDate;

    private Date endDate;

    @I18Constraint(value = "patterns.zip")
    private String zip;

    @AnotherI18NConstraint(value = "patterns.zip")
    private String anotherZip;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Boolean getDone() {
        return done;
    }

    public void setDone(Boolean done) {
        this.done = done;
    }

    public Date getDueDate() {
        return dueDate;
    }

    public void setDueDate(Date dueDate) {
        this.dueDate = dueDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }

    public String getZip() {
        return zip;
    }

    public void setZip(String zip) {
        this.zip = zip;
    }

    public String getAnotherZip() {
        return anotherZip;
    }

    public void setAnotherZip(String anotherZip) {
        this.anotherZip = anotherZip;
    }

}
